### ISSUE TITLE
WebHook http_request_failed

### ISSUE THREAD

**Issue by tanguy-opendsi** [woocommerce/woocommerce#29035]
Hi, 
i'm using woosync but when i'm using hook i got
invalid_url 
I found this tutorial and it's work, but i dislike edit de core code... [https://stackoverflow.com/questions/31875442/woocommerce-webhook-invalid-url](https://stackoverflow.com/questions/31875442/woocommerce-webhook-invalid-url)
woocommerce use wp_sage_remote_request.
And the security here failed because i'm talking with a non public ip server (LAN)
It's possible to add reject_unsafe_urls inside woocommerce config 
Best regards

**Comment by barryhughes** [comment]
Thanks for reaching out, @tanguy-opendsi.

I agree that modifying core code is generally not a good idea, but you don't actually have to. Instead, you could add some code to a custom plugin along these lines:

```php
function http_additional_safe_urls( $http_request_args, $url ) {
	// Do not interfere unless the URL matches a pattern we trust.
	// You could expand this to match against multiple strings, or
	// even use a regular expression, depending on your needs.
	if ( $url !== 'http://localhost:3333' ) {
		return $http_request_args;
	}

	// Disable the `reject_unsafe_urls` arg.
	$http_request_args['reject_unsafe_urls'] = false;
	return $http_request_args;
};

add_filter('http_request_args', 'http_additional_safe_urls', 10, 2 );
```

You would need to modify slightly to suit your specific needs, but the essence of this is that it makes an exception (in my case, for calls to `localhost:3000`—but you could certainly adapt to test against different or multiple URLs) so that you can access URLs that normally would be rejected by `wp_safe_remote_get()`.

In short, so far as I can tell, we're not looking at a bug or problem in WooCommerce: instead it's a matter of making exceptions to WordPress's own safety precautions (and it makes sense to do this on a case-by-case basis as and when needed).

Let me know if you have thoughts/concerns, otherwise we'll go ahead and close this issue.

**Comment by KnowYourselves** [comment]
Hello @barryhughes!

I'm having the same problem. I have a local instance of woocommerce (following [this guide](https://github.com/woocommerce/woocommerce/wiki/How-to-set-up-WooCommerce-development-environment)) and I can't get the webhooks to work with local urls (for testing).

I understand the code you have shared, and I kind of get why it should work, but I'm really new to the world of wordpress so I don't really understand where I should add that type of code. Could you perhaps explain a little further?, it would help a lot.

Cheers!

**Comment by barryhughes** [comment]
Hi @KnowYourselves—so long as you are positive that the underlying problem is the same, you could modify the snippet I shared as needed and then add it as a [custom plugin](https://developer.wordpress.org/plugins/) (and there may be other solutions, too, but this way is pretty standard).

Just to set expectations, though, I do want to quickly note [we cannot provide support](https://github.com/woocommerce/woocommerce/blob/b6f3c2b0bee2cf23321a511010657612cf66e8c5/README.md) here in the issue tracker (bear in mind that this problem is not really specific to WooCommerce, but instead relates to WordPress development in general):

> This repository is not suitable for support. Please don't use our issue tracker for support requests, but for core WooCommerce issues only. Support can take place through the appropriate channels:
>
> - If you have a problem, you may want to start with the self help guide.
> - The WooCommerce.com premium support portal for customers who have purchased themes or extensions.
> - Our community forum on wp.org which is available for all WooCommerce users.
> - The WooCommerce Help and Share Facebook group.
> - For customizations, you may want to check our list of WooExperts or Codeable.

Thanks!
